releng/third-party/pom.xml

```diff
@@ -57,11 +59,12 @@
 		<lz4.version>1.8.0</lz4.version>
 		<hdrhistogram.version>2.1.12</hdrhistogram.version>
 		<jemmy.version>2.0.0</jemmy.version>
-		<jetty.version>10.0.18</jetty.version>
+		<jetty.version>12.0.6</jetty.version>
```

Comment:
This is where I think it gets a bit difficult for backwards compatibility. Because we're putting the jetty dependencies into the p2 repository, the changes to this pom affect all the platform definitions we use. So now that this is updated to use 12.0.6 and pull in the jetty ee9 dependencies (which are the 12.X.X), 2023-12 can't build because it's unable to find the 10.0.18 versions of the jetty-related dependencies.

This is where I was originally taking a look but didn't get too far, but would it be possible to use the jetty dependencies that are shipped by Eclipse instead ([link](https://download.eclipse.org/releases/2024-03/202403131000/plugins/))? And this way each platform-definition we have could just reference the same version that is shipped by that Eclipse release?

Reply:
For the class **WebsocketServer.java**, we need **org.eclipse.jetty.websocket.servlet** (Jetty 10.0.18) or **org.eclipse.jetty.ee9.websocket.servle**t (Jetty 12.0.6). These are not present in https://download.eclipse.org/releases/2024-03/202403131000/plugins/.

Reply:
> This is where I think it gets a bit difficult for backwards compatibility. Because we're putting the jetty dependencies into the p2 repository, the changes to this pom affect all the platform definitions we use. So now that this is updated to use 12.0.6 and pull in the jetty ee9 dependencies (which are the 12.X.X), 2023-12/2023-09/2023-03 can't build because it's unable to find the 10.0.18 versions of the jetty-related dependencies.
> 
> This is where I was originally taking a look but didn't get too far, but would it be possible to use the jetty dependencies that are shipped by Eclipse instead ([link](https://download.eclipse.org/releases/2024-03/202403131000/plugins/))? And this way each platform-definition we have could just reference the same version that is shipped by that Eclipse release?
For the class WebsocketServer.java, we need org.eclipse.jetty.websocket.servlet (Jetty 10.0.18) or org.eclipse.jetty.ee9.websocket.servlet (Jetty 12.0.6). These are not present in https://download.eclipse.org/releases/2024-03/202403131000/plugins/.

Reply:
> This is where I think it gets a bit difficult for backwards compatibility. Because we're putting the jetty dependencies into the p2 repository, the changes to this pom affect all the platform definitions we use. So now that this is updated to use 12.0.6 and pull in the jetty ee9 dependencies (which are the 12.X.X), 2023-12/2023-09/2023-03 can't build because it's unable to find the 10.0.18 versions of the jetty-related dependencies.
> 
> This is where I was originally taking a look but didn't get too far, but would it be possible to use the jetty dependencies that are shipped by Eclipse instead ([link](https://download.eclipse.org/releases/2024-03/202403131000/plugins/))? And this way each platform-definition we have could just reference the same version that is shipped by that Eclipse release?



For the class WebsocketServer.java, we need org.eclipse.jetty.websocket.servlet (Jetty 10.0.18) or org.eclipse.jetty.ee9.websocket.servlet (Jetty 12.0.6). These are not present in https://download.eclipse.org/releases/2024-03/202403131000/plugins/.

Reply:
And I think this problem was there from the beginning not only for jetty but for other libraries too. Or may be we can have separate **third-party/pom.xml** for each **target platform**.

Reply:
> For the class **WebsocketServer.java**, we need **org.eclipse.jetty.websocket.servlet (Jetty 10.0.18)** or **org.eclipse.jetty.ee9.websocket.servlet (Jetty 12.0.6)**. These are not present in https://download.eclipse.org/releases/2024-03/202403131000/plugins/.

Ah right, that's unfortunate. In response to Marcus' comment on Slack maybe it's best to do away with trying to keep all these platforms and just maintain the most current version.